modules/perception/camera/tools/offline/visualizer.cc

```diff
@@ -103,33 +104,78 @@ bool Visualizer::Init(const std::vector<std::string> &camera_names,
 
 bool Visualizer::Init_all_info_single_camera(
     const std::string &camera_name,
-    std::map<std::string, Eigen::Matrix3f> intrinsic_map,
-    std::map<std::string, Eigen::Matrix4d> extrinsic_map,
-    Eigen::Matrix4d ex_lidar2imu, double pitch_adj, int image_height,
-    int image_width) {
+    const std::map<std::string, Eigen::Matrix3f> &intrinsic_map,
+    const std::map<std::string, Eigen::Matrix4d> &extrinsic_map,
+    const Eigen::Matrix4d &ex_lidar2imu,
+    const double &pitch_adj_degree,
+    const double &yaw_adj_degree,
+    const double &roll_adj_degree,
+    const int image_height,
+    const int image_width) {
   image_height_ = image_height;
   image_width_ = image_width;
   intrinsic_map_ = intrinsic_map;
   extrinsic_map_ = extrinsic_map;
+  ex_lidar2imu_ = ex_lidar2imu;
+
   last_timestamp_ = 0;
   small_h_ = static_cast<int>(image_height_ * scale_ratio_);
   small_w_ = static_cast<int>(image_width_ * scale_ratio_);
   world_h_ = 2 * small_h_;
 
+  AINFO << "world_h_: " << world_h_;
+  AINFO << "wide_pixel_: " << wide_pixel_;
+  AINFO << "small_h_: " << small_h_;
+  AINFO << "small_w_: " << small_w_;
   camera_image_[camera_name + "_2D"] =
       cv::Mat(small_h_, small_w_, CV_8UC3, cv::Scalar(0, 0, 0));
   camera_image_[camera_name + "_3D"] =
       cv::Mat(small_h_, small_w_, CV_8UC3, cv::Scalar(0, 0, 0));
   world_image_ = cv::Mat(world_h_, wide_pixel_, CV_8UC3, cv::Scalar(0, 0, 0));
   draw_range_circle();
 
-  ex_lidar2imu.block(0, 3, 3, 1) = -ex_lidar2imu.block(0, 3, 3, 1);
-  extrinsic_map_.at(camera_name).block(0, 3, 3, 1) =
-      -extrinsic_map_.at(camera_name).block(0, 3, 3, 1);
-  homography_im2car_ = Camera2CarHomograph(
-      intrinsic_map_.at(camera_name).cast<double>(),
-      extrinsic_map_.at(camera_name), ex_lidar2imu, pitch_adj);
-  AINFO << "homography_im2car_: " << homography_im2car_;
+  // 1. transform camera_>lidar
+  ex_camera2lidar_ = extrinsic_map_.at(camera_name);
+  AINFO << "ex_camera2lidar_ = ";
+  AINFO << extrinsic_map_.at(camera_name);
+
+  AINFO << "ex_lidar2imu_ =";
+  AINFO << ex_lidar2imu_;
+
+  // 2. transform camera->lidar->imu
+  ex_camera2imu_ = ex_lidar2imu_ * ex_camera2lidar_;
+  AINFO << "ex_camera2imu_ =";
+  AINFO << ex_camera2imu_;
+
+  // intrinsic camera parameter
+  K_ = intrinsic_map_.at(camera_name).cast<double>();
+  AINFO << "intrinsic K_ =";
+  AINFO << K_;
+  // homography_ground2image_.setIdentity();
+  // homography_image2ground_.setIdentity();
+
+  // rotate 90 degree around z axis to make x point forward
+  // double imu_height = 0;  // imu height should be considred later
+  ex_imu2car_ << 0, 1, 0, 0,  // cos(90), sin(90), 0,
+                -1, 0, 0, 0,  // -sin(90),  cos(90), 0,
+                0, 0, 1, 0,  // 0,              0, 1
+                0, 0, 0, 1;
+
+  // 3. transform camera->lidar->imu->car
+  ex_camera2car_ = ex_imu2car_ * ex_camera2imu_;
+
+  AINFO << "ex_camera2car_ =";
+  AINFO << ex_camera2car_;
```

Comment:
Please concat them:
AINFO << "ex_camera2car_ =" << ex_camera2car_;

as well as other places.

Reply:
Will merge all separate AINFOs to one.Thank you!